common/vcomputer-system.cpp

```diff
@@ -2,146 +2,137 @@
-				std::unique_ptr <TR3200> trcpu = std::make_unique<TR3200>();
-				this->vc.SetCPU(std::move(trcpu));
-				this->vc.On();
-				this->vc.SetState(&state, sizeof(TR3200State));
-			}
-			break;
-			case proto::Computer::CPU::kDcpu16N:
-			{
-				const proto::Computer::CPU::DCPU16N& dcpu16n = cpu.dcpu16n();
-				DCPU16NState state;
-				for (int i = 0; i < dcpu16n.registers_size(); ++i) {
-					state.r[i] = dcpu16n.registers(i);
-				}
-				state.pc = dcpu16n.pc();
-				state.wait_cycles = dcpu16n.wait_cycles();
-				std::unique_ptr <DCPU16N> dcpucpu = std::make_unique<DCPU16N>();
-				this->vc.SetCPU(std::move(dcpucpu));
-				this->vc.On();
-				this->vc.SetState(&state, sizeof(DCPU16NState));
-			}
-			break;
-			case proto::Computer::CPU::CPU_NOT_SET:
-			break;
+Computer::Computer() : rom(new std::uint8_t[32 * 1024]), rom_size(0) {}
 
+void Computer::In(const proto::Component& source) {
+	const proto::Computer& comp = source.computer();
+	const proto::Computer::CPU& cpu = comp.cpu();
+	switch (cpu.cpu_case()) {
+	case proto::Computer::CPU::kTr3200:
+	{
+		const proto::Computer::CPU::TR3200& tr3200 = cpu.tr3200();
+		TR3200State state;
+		for (int i = 0; i < tr3200.registers_size(); ++i) {
+			state.r[i] = tr3200.registers(i);
 		}
-		std::string buf = comp.ram();
-		trillek::Byte* ram = this->vc.Ram();
-		for (std::size_t i = 0; i < buf.length(); ++i) {
-			ram[i] = buf[i];
-		}
-		for (int i = 0; i < comp.devices_size(); ++i) {
-			const proto::Computer::Device& device = comp.devices(i);
-			switch (device.device_case()) {
-				case proto::Computer::Device::kComputerScreen:
-				{
-					std::shared_ptr<ComputerScreen> screen = std::make_shared<ComputerScreen>();
-					screen->In(device);
-					this->vc.AddDevice(device.slot(), screen->device);
-					this->devices[device.slot()] = screen;
-				}
-				break;
-				case proto::Computer::Device::kComputerKeyboard:
-				{
-					std::shared_ptr<ComputerKeyboard> keybaord = std::make_shared<ComputerKeyboard>();
-					keybaord->In(device);
-					this->vc.AddDevice(device.slot(), keybaord->device);
-					this->devices[device.slot()] = keybaord;
-				}
-				break;
-				case proto::Computer::Device::DEVICE_NOT_SET:
-				break;
-			}
-		}
-
-		std::string fname{ comp.rom_file() };
-		int size = LoadROM(FilePath::GetAssetPath(fname).toString(), this->rom);
-		if (size < 0) {
-			std::shared_ptr<spdlog::logger> _log = spdlog::get("console_log");
-			_log->error("An error hapen when was reading the file {}", fname);
+		state.pc = tr3200.pc();
+		state.wait_cycles = tr3200.wait_cycles();
+		state.int_msg = static_cast<trillek::Word>(tr3200.int_msg());
+		state.interrupt = tr3200.interrupt();
+		state.step_mode = tr3200.step_mode();
+		state.skiping = tr3200.skiping();
+		state.sleeping = tr3200.sleeping();
+		std::unique_ptr<TR3200> trcpu = std::make_unique<TR3200>();
+		this->vc.SetCPU(std::move(trcpu));
+		this->vc.On();
+		this->vc.SetState(&state, sizeof(TR3200State));
+	} break;
```

Comment:
again, the `break`s should go in the braces

Reply:
fixed